ezomero/json_api.py

```diff
@@ -0,0 +1,248 @@
+        * OMERO_PASS
+        * OMERO_WEB_HOST
+
+    3) If environment variables are not set, try to load from a config file.
+       This file should be called '.ezomero'. By default, this function will
+       look in the home directory, but ``config_path`` can be used to specify
+       a directory in which to look for '.ezomero'.
+
+       The function ``ezomero.store_connection_params`` can be used to create
+       the '.ezomero' file.
+
+       Note that passwords can not be loaded from the '.ezomero' file. This is
+       to discourage storing credentials in a file as cleartext.
+
+    4) If any remaining parameters have not been set by the above steps, the
+       user is prompted to enter a value for each unset parameter.
+    """
+    # load from .ezomero config file if it exists
+    if config_path is None:
+        config_fp = Path.home() / '.ezomero'
+    elif type(config_path) is str:
+        config_fp = Path(config_path) / '.ezomero'
+    else:
+        raise TypeError('config_path must be a string')
+
+    config_dict = {}
+    if config_fp.exists():
+        config = configparser.ConfigParser()
+        with config_fp.open() as fp:
+            config.read_file(fp)
+        try:
+            config_dict = config["JSON"]
+        except KeyError:
+            raise KeyError('.ezomero does not contain JSON information.')
+
+    # set user
+    if user is None:
+        user = config_dict.get("OMERO_USER", user)
+        user = os.environ.get("OMERO_USER", user)
+    if user is None:
+        user = input('Enter username: ')
+
+    # set password
+    if password is None:
+        password = os.environ.get("OMERO_PASS", password)
+    if password is None:
+        password = getpass('Enter password: ')
+
+    # set web host
+    if web_host is None:
+        web_host = config_dict.get("OMERO_WEB_HOST", web_host)
+        web_host = os.environ.get("OMERO_WEB_HOST", web_host)
+    if web_host is None:
+        web_host = input('Enter host: ')
+
+    session = requests.Session()
+    # Start by getting supported versions from the base url...
+    api_url = '%s/api/' % web_host
+    r = session.get(api_url, verify=verify)
+    # we get a list of versions
+    versions = r.json()['data']
+    # use most recent version...
+    version = versions[-1]
+    # get the 'base' url
+    base_url = version['url:base']
+    r = session.get(base_url)
+    # which lists a bunch of urls as starting points
+    urls = r.json()
+    servers_url = urls['url:servers']
+    login_url = urls['url:login']
+
+    # To login we need to get CSRF token
+    token_url = urls['url:token']
+    token = session.get(token_url).json()['data']
+    # We add this to our session header
+    # Needed for all POST, PUT, DELETE requests
+    session.headers.update({'X-CSRFToken': token,
+                            'Referer': login_url})
+
+    # List the servers available to connect to
+    servers = session.get(servers_url).json()['data']
+
+    SERVER_NAME = 'omero'
```

Comment:
is this meant to be a constant? If so it should be assigned at the module level, else it should be snake_case

Reply:
also, will the server name always be 'omero'? Is that something that could change? Should it be a parameter?

Reply:
I think this could be a parameter - this was boilerplate code from OME that I did not look into thoroughly enough, apparently